File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        l,r = 0,0
        vowels = set(["a","e","i","o","u"])
        ret =0
        curr = 0
        while(r<len(s)):
            if s[r].lower() in vowels:
                curr+=1
            if r-l==k:
                curr = curr if s[l].lower() not in vowels else curr-1
                l+=1
            ret = max(ret,curr)
            r+=1

        return ret

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.ts
function maxVowels(s: string, k: number): number {
    let hashmap = new Map<string,number>()
    hashmap.set("a",0)
    hashmap.set("e",0)
    hashmap.set("i",0)
    hashmap.set("o",0)
    hashmap.set("u",0)
    let ret = 0
    let count = 0
    let l = 0
    for (let r = 0;r<s.length;r++){
        if (hashmap.has(s[r].toLowerCase()) ){
            hashmap.set(s[r].toLowerCase(),hashmap.get(s[r].toLowerCase())+1)
            count++;
        }
        if (r-l+1 ==k){
            ret = Math.max(ret,count)
            if (hashmap.has(s[l].toLowerCase())){
                hashmap.set(s[l].toLowerCase(),hashmap.get(s[l].toLowerCase())-1) 
                count--
            }
            l++
        }
    }
    return ret
};